src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachToCaseCallbackService.java

```diff
@@ -118,13 +121,19 @@ public AttachToCaseCallbackService(
 
         return getValidation(exceptionRecordDetails, requesterIdamToken, requesterUserId)
             .map(callBackEvent -> tryAttachToCase(callBackEvent, exceptionRecordDetails, ignoreWarnings))
-            .map(attachToCaseResult ->
-                attachToCaseResult.map(modifiedFields ->
-                    mergeCaseFields(exceptionRecordDetails.getData(), modifiedFields))
-            )
+            .map(errorsOrRef -> errorsOrRef.map(caseRef -> finalizeExceptionRecordData(exceptionRecordDetails, caseRef)))
```

Comment:
with the change of argument i can see validation/eithers are not used properly :(. perhaps for later review

Reply:
Here's a revised usage of this section ;):

```java
return getValidation(exceptionRecordDetails, requesterIdamToken, requesterUserId)
    .toEither()
    .bimap(
        errors -> ErrorsAndWarnings.withErrors(errors.asJava()),
        callBackEvent -> tryAttachToCase(callBackEvent, exceptionRecordDetails, ignoreWarnings)
    )
    .flatMap(Function.identity())
    .map(targetCaseRef -> finalizeExceptionRecordData(exceptionRecordDetails, targetCaseRef));
```

Reply:
👍 I'll look into that on a separate PR